base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1771,10 +1813,15 @@ fn reorganize_chain<T: BlockchainBackend>(
     backend: &mut T,
     block_validator: &dyn PostOrphanBodyValidation<T>,
     height: u64,
+    current_tip_height: u64,
     chain: &VecDeque<Arc<ChainBlock>>,
 ) -> Result<Vec<Arc<ChainBlock>>, ChainStorageError>
 {
-    let removed_blocks = rewind_to_height(backend, height)?;
+    let removed_blocks = if height < current_tip_height {
```

Comment:
Should this not be "<=". 

If the height is the same, one block should be removed.